### ISSUE TITLE
Running mvn package on Xtext parent project with Xtext 2.17.0 produces IllegalArgumentException

### ISSUE THREAD

**Issue by scanyboss** [eclipse/xtext-core#1120]
IllegalArgumentException is thrown after running mvn package on the Xtext parent project generated by Xtext 2.17.0 programmatically. 

How to reproduce: 
1. download https://github.com/scanyboss/hello-world-language project
2. run mvn compile 
3. run Main.main() method - Code contains the path to the maven, as the mvn compile is run programmatically - needs to be changed! 
4. run mvn package on the Xtext parent project generated by the main method. 
The last step produces the error.

**Comment by cdietrich** [comment]
hmmmm in my try the last one does not fail.
can you please checkin the result of (3) too

**Comment by scanyboss** [comment]
I'm using Fedora 27. 
I have no problem with step 3. 
![Screenshot from 2019-04-09 21-14-33](https://user-images.githubusercontent.com/17799565/55828544-f46c8a80-5b0c-11e9-98f5-01fb7c216a00.png)

But when I was running maven action programmatically for the first time I had to add execute permissions to the maven itself because even when the maven action doesnt fail, "permission denied" can be printed in the logs. Dont you see it when running step 3 ?

**Comment by cdietrich** [comment]
no, i have the problem: all 4 steps run smoothly

```
INFO] Reactor Summary for org.hello.parent 1.0.0-SNAPSHOT:
[INFO] 
[INFO] org.hello.parent ................................... SUCCESS [  0.007 s]
[INFO] org.hello .......................................... SUCCESS [ 20.122 s]
[INFO] org.hello.ide ...................................... SUCCESS [  2.589 s]
[INFO] org.hello.ui ....................................... SUCCESS [  3.158 s]
[INFO] org.hello.target ................................... SUCCESS [  0.058 s]
[INFO] ------------------------------------------------------------------------
```
=> to verify your failure at (4) i want to start at the end of (3)

**Comment by scanyboss** [comment]
Are you running it directly in Eclipse IDE ?

**Comment by cdietrich** [comment]
no on bash

**Comment by scanyboss** [comment]
okay I tried it also in terminal and it works there. the problem is that when you look at the width of the line, in the successful case each line contains 2 characters less than the unsuccessful case (which occurs in IDE). Take a look at: https://www.diffchecker.com/AAHDEoqP ... that 2 characters on each line cause that on the some line is the bundle-version broken in the middle - and therefore manifest merger write the bundle-version to this line once again - that means it will be written twice within a particular item or package. And that's why it's happening ... because in successful case no bundle-version is broken by linebreak in the middle ... so even if there are some packages written twice - that is not a problem. The problem is just when there are 2 bundle-versions for the same package.

**Comment by scanyboss** [comment]
IDK why there are 2 characters more/less on the particular line ... can be caused by line delimiter maybe. But the problem is that it's a little bit coincidence and it depends on what you write to the mainfest and in what order ... anytime when happen that bundle-version is broken by linebreak it will be written to the manifest once again and then the problem occurs.

**Comment by cdietrich** [comment]
I still don’t understand that. Who formats that file.
If you check it in the way it is initially created then the problem should 
Happen regardless if you call it from eclipse or from command line
I want to have a project in a state where I can verify the prolem.

If the merge algorithm is buggy I want to see
This file is merged and it breaks

**Comment by cdietrich** [comment]
So the problem is before (4) or after (4) and the manifest differs after (4) from eclipse and from command line?

**Comment by scanyboss** [comment]
In (3) I'm changing the manifest file (see the Main - line 59). When I change it this way there are no linebreaks anymore. Then in (4) when the merger modifies the manifest and finds that "bundle-version" is split into 2 lines by linebreak another "bundle-version" is added to this particular package. So the problem is that in (4) the merger appends wrong data in case of "bundle-version" is split into 2 lines. 

Also question is that why there are no line breaks after my change. Should I add them somehow explicitely ?

**Comment by cdietrich** [comment]
No please share the manifest after your change

**Comment by scanyboss** [comment]
`Manifest-Version: 1.0
Bundle-ManifestVersion: 2
Bundle-Name: org.hello.ui
Bundle-Vendor: My Company
Bundle-Version: 1.0.0.qualifier
Bundle-SymbolicName: org.hello.ui; singleton:=true
Bundle-ActivationPolicy: lazy
Require-Bundle: org.hello,org.hello.ide,org.eclipse.xtext.ui,org.eclipse
 .xtext.ui.shared,org.eclipse.xtext.ui.codetemplates.ui,org.eclipse.ui.e
 ditors;bundle-version="3.5.0",org.eclipse.ui.ide;bundle-version="3.5.0"
 ,org.eclipse.ui,org.eclipse.compare,org.eclipse.xtext.builder,org.eclip
 se.xtext.xbase.lib;bundle-version="2.14.0",org.eclipse.xtend.lib;bundle
 -version="2.14.0";resolution:=optional,org.eclipse.ui.console;bundle-ve
 rsion="3.8.100"
Import-Package: org.apache.log4j
Bundle-RequiredExecutionEnvironment: JavaSE-1.8
Automatic-Module-Name: org.hello.ui
Export-Package: org.hello.ui.internal,org.hello.ui.contentassist,org.hel
 lo.ui.quickfix
Bundle-Activator: org.hello.ui.internal.HelloActivator

`

**Comment by cdietrich** [comment]
@ArneDeutsch do you have any idea yet?

**Comment by cdietrich** [comment]
@ArneDeutsch any news on this?

**Comment by cdietrich** [comment]
@ArneDeutsch ping

**Comment by ArneDeutsch** [comment]
I executed the steps outlined in the ticket with current head version. I adapted the path to maven installation (used 3.5.0) and changed Xtext version to 2.19.0 (with 2.17.0 I get an issue with java 11 ... "Fatal error compiling: invalid target release: 11"). With this settings I can execute all steps above without error. I suspect the issue was fixed for 2.18.0 or 2.19.0 (I remember there were some fixes). @scanyboss can you try if it is working for you for more recent Xtext version?

**Comment by ArneDeutsch** [comment]
Correcting myself ... have it running with Xtext 2.17.0 as well. Works also. Also did some experiments with provided manifests (applying manifest merging logic against it, also in slightly modified versions .... but I can not detect any misbehaviour here.

**Comment by cdietrich** [comment]
I could not reproduce this neither

**Comment by scanyboss** [comment]
I've just tried on both (2.17.0 and 2.19.0) versions and got this error every time (run from terminal on ubuntu 18.04).

**Comment by cdietrich** [comment]
i wonder how to exactly reproduce this then.
how to get the generated project in a state you have it.
i tried this on linux and it works fine

**Comment by cdietrich** [comment]
i can reproduce the manifest in ui project looks like

```
Require-Bundle: org.hello,org.hello.ide,org.eclipse.xtext.ui,org.eclip
 se.xtext.ui.shared,org.eclipse.xtext.ui.codetemplates.ui,org.eclipse.
 ui.editors;bundle-version="3.5.0",org.eclipse.ui.ide;bundle-version="
 3.5.0",org.eclipse.ui,org.eclipse.compare,org.eclipse.xtext.builder,o
 rg.eclipse.xtext.xbase.lib;bundle-version="2.14.0",org.eclipse.xtend.
 lib;bundle-version="2.14.0";resolution:=optional,org.eclipse.ui.conso
 le;bundle-version="3.8.100"
```

after running the main though. can you @ArneDeutsch ?

**Comment by scanyboss** [comment]
Yes, but look at mine: 
![Screenshot from 2019-09-09 18-34-59](https://user-images.githubusercontent.com/17799565/64549261-a6181d80-d330-11e9-8ffe-e40e65206f08.png)

it's little bit different ... I have 2 more characters in the first line, even when I dont know why.

**Comment by cdietrich** [comment]
the question is: if you change your main non to run maven, but to create the project only. 
does this problem happen too if you create the project and then run mvn on command line twice?

**Comment by cdietrich** [comment]
=> interesting would be how the manifest looks like before you write it in the main

**Comment by scanyboss** [comment]
> the question is: if you change your main non to run maven, but to create the project only.
does this problem happen too if you create the project and then run mvn on command line twice?
(if needed run a second main that changes the manifest in between)

 Yes the same problem happens in this case. 

> interesting would be how the manifest looks like before you write it in the main

Like this:
```
Manifest-Version: 1.0
Bundle-ManifestVersion: 2
Bundle-Name: org.hello.ui
Bundle-Vendor: My Company
Bundle-Version: 1.0.0.qualifier
Bundle-SymbolicName: org.hello.ui; singleton:=true
Bundle-ActivationPolicy: lazy
Require-Bundle: org.hello,
 org.hello.ide,
 org.eclipse.xtext.ui,
 org.eclipse.xtext.ui.shared,
 org.eclipse.xtext.ui.codetemplates.ui,
 org.eclipse.ui.editors;bundle-version="3.5.0",
 org.eclipse.ui.ide;bundle-version="3.5.0",
 org.eclipse.ui,
 org.eclipse.compare,
 org.eclipse.xtext.builder,
 org.eclipse.xtext.xbase.lib;bundle-version="2.14.0",org.eclipse.xtend.lib;bundle-version="2.14.0";resolution:=optional
Import-Package: org.apache.log4j
Bundle-RequiredExecutionEnvironment: JavaSE-1.8
Automatic-Module-Name: org.hello.ui
Export-Package: org.hello.ui.internal,
 org.hello.ui.contentassist,
 org.hello.ui.quickfix
Bundle-Activator: org.hello.ui.internal.HelloActivator
```

**Comment by cdietrich** [comment]
does this happen if you run the workflow manually and not maven too?

**Comment by scanyboss** [comment]
In general there is a restriction that manifest.mf file can contain only lines with max 72 bytes. All the chars in the first line are encoded with 1byte (UTF-8) -> that means after 72 chars, all the remaining chars are put to the next line. I wonder why your first line has only 70 chars ? Which encoding are you using ?

**Comment by cdietrich** [comment]
i did not set any specific encoding. my eclipses encoding is UTF-8.
i am not sure if this 72byte things is reflected somewhere in the code and or wizard.

where do you have the 72byte information from? which exact jdk do you use?

**Comment by cdietrich** [comment]
"Require-Bundle: org.hello,org.hello.ide,org.eclipse.xtext.ui,org.eclip" has 70 chars though

**Comment by scanyboss** [comment]
https://stackoverflow.com/questions/28727048/why-jar-manifest-file-shouldnt-contain-lines-with-more-than-72-bytes -> it's pretty old post but maybe still valid 
I'm using jdk 11.0.4

**Comment by cdietrich** [comment]
i am using java 11. does this happen in java 8 too?

**Comment by cdietrich** [comment]
i really wonder if this is related to you using the Manifest class (and not the MergableManifest2 class)

**Comment by cdietrich** [comment]
in my manifest the line ending seems to be `\r\n`

**Comment by cdietrich** [comment]
running your main with java 11 indeed gives

```
Manifest-Version: 1.0
Bundle-ManifestVersion: 2
Bundle-Name: org.hello.ui
Bundle-Vendor: My Company
Bundle-Version: 1.0.0.qualifier
Bundle-SymbolicName: org.hello.ui; singleton:=true
Bundle-ActivationPolicy: lazy
Require-Bundle: org.hello,org.hello.ide,org.eclipse.xtext.ui,org.eclipse
 .xtext.ui.shared,org.eclipse.xtext.ui.codetemplates.ui,org.eclipse.ui.e
 ditors;bundle-version="3.5.0",org.eclipse.ui.ide;bundle-version="3.5.0"
 ,org.eclipse.ui,org.eclipse.compare,org.eclipse.xtext.builder,org.eclip
 se.xtext.xbase.lib;bundle-version="2.14.0",org.eclipse.xtend.lib;bundle
 -version="2.14.0";resolution:=optional,org.eclipse.ui.console;bundle-ve
 rsion="3.8.100"
Import-Package: org.apache.log4j
Bundle-RequiredExecutionEnvironment: JavaSE-1.8
Automatic-Module-Name: org.hello.ui
Export-Package: org.hello.ui.internal,org.hello.ui.contentassist,org.hel
 lo.ui.quickfix
Bundle-Activator: org.hello.ui.internal.HelloActivator


```

**Comment by cdietrich** [comment]
@ArneDeutsch did you test that mergablemanifest2 makes with this?

**Comment by scanyboss** [comment]
Seems like MergeableManifest2 works fine.

**Comment by cdietrich** [comment]
we should look into that problem anyway. from my pov we should be able to deal with manifest as java 11 creates it too

**Comment by ArneDeutsch** [comment]
Identified the issue. Case is when the "bundle-version" identifier contains linebreaks. Fix commited.

**Comment by cdietrich** [comment]
Fixed in 2.20
